### PR TITLE
Updated dashboard recent activities to use ant design

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Updated the dashboard to use ant design.

**Admin -> Recent Activity for admin's projects**

admindashboarduserprojects

**Admin -> Recent Activity for all projects**

admindashboardallprojects

**User -> Recent Activity for users's projects**

userdashboard

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [ ] User documentation updated for UI or technical changes.

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/repositories/joins/project/UserGroupProjectJoinRepository.java [hunk 1]
```diff
@@ -75,4 +75,5 @@
 	 * @return the relationship between the group and project.
 	 */
 	public UserGroupProjectJoin findByProjectAndUserGroup(final Project p, final UserGroup userGroup);
+
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/repositories/joins/project/UserGroupProjectJoinRepository.java:78]: I think you commited this by mistake?

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/activities/ActivityType.java [hunk 2]
```diff
@@ -9,6 +9,7 @@
 	PROJECT_USER_ROLE("project_user_role_updated"),
 	PROJECT_USER_REMOVED("project_user_removed"),
 	PROJECT_SAMPLE_ADDED("project_sample_added"),
+	PROJECT_SAMPLE_REMOVED("project_sample_removed"),
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/activities/ActivityType.java:12]: What is this here for?

**Comment by joshsadam** [comment]
Here is a long standing frustration of mine, and I probably caused it.  We have both "events" and "activities" and I am not really sure what the difference is between them (but I think they are the same thing.  We need to do better at naming things consistently.

File: src/main/java/ca/corefacility/bioinformatics/irida/service/impl/analysis/submission/AnalysisSubmissionServiceImpl.java [hunk 3]
```diff
@@ -831,4 +831,12 @@ public Long getAnalysesRanInTimePeriod(Date createdDate) {
 		return analysisSubmissionRepository.countAnalysesRanGrouped(createdDate, statisticTimePeriod.getGroupByFormat());
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	@PreAuthorize("hasRole('ROLE_ADMIN') or hasRole('ROLE_USER')")
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/service/impl/analysis/submission/AnalysisSubmissionServiceImpl.java:837]: Isn't ROLE_ADMIN just a super set of the ROLE_USER?  Do you need to authorize both?

File: src/main/java/ca/corefacility/bioinformatics/irida/service/impl/ProjectServiceImpl.java [hunk 4]
```diff
@@ -502,6 +502,26 @@ public void removeSamplesFromProject(Project project, Iterable<Sample> samples)
 				.build();
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	@PreAuthorize("hasRole('ROLE_USER')")
+	public List<Project> getProjectsForUserUnique(User user) {
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/service/impl/ProjectServiceImpl.java:511]: You might want to talk to Katherine about these, as they might be useful for her work with the user > projects tab on her account refresh.

File: src/main/webapp/resources/js/apis/activities/activities.js [hunk 5]
```diff
@@ -22,3 +22,31 @@ export function getProjectActivities({ projectId, page = 0 }) {
...
+ * Get a page of recent activities for all projects
+ *
+ * @param {number} page - page of activities requested
+ * @returns {Promise<AxiosResponse<any>>}
+ */
+export function getAllRecentActivities({ page = 0 }) {
+  try {
+    return axios.get(`${BASE_URL}/all?page=${page}`).then(({ data }) => data);
+  } catch (e) {
+    return Promise.reject(i18n("RecentActivity.loadError"));
```
Comment (joshsadam) [inline src/main/webapp/resources/js/apis/activities/activities.js:50]: Why aren't you returning the error message from the server?  Saves client side templating

File: src/main/webapp/resources/js/pages/dashboard/components/RecentActivity.jsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import { Button, Space, Typography } from "antd";
+import React from "react";
+import { grey1 } from "../../../styles/colors";
+import { SPACE_MD } from "../../../styles/spacing";
+import { RecentActivityAllProjects } from "./RecentActivityAllProjects";
+import { RecentActivityUserProjects } from "./RecentActivityUserProjects";
+
+const isAdmin = window.TL._USER.systemRole === "ROLE_ADMIN";
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/dashboard/components/RecentActivity.jsx:8]: There is an `isAdamin` function in `role-utilities.js` that handles tihs better.  I still hate that it checkes the `window.TL` variable, we need to eventually get rid of that.

File: src/main/webapp/resources/js/pages/dashboard/components/RecentActivity.jsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+ * Component to display recent activity
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export function RecentActivity() {
+  const [allProjectsVisible, setAllProjectsVisible] = React.useState(false);
+  const [userProjectsVisible, setUserProjectsVisible] = React.useState(true);
+
+  return (
+    <Space
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/dashboard/components/RecentActivity.jsx:20]: Space does work good here, lately I have been findign that using there Grid system actually works slightly better.  Wrap each in a `Col` tag and Wrap everything in a `<Row gutter={[16,16]}>` seems to handle it very well.

File: src/main/webapp/resources/js/pages/dashboard/components/RecentActivityAllProjects.jsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        const list = addKeysToList(data.content, "activity", "date");
+        setActivities([...activities, ...list]);
+        setTotal(data.total);
+        setLoading(false);
+      })
+      .catch((error) => notification.error({ message: error }));
+  }, [page]);
+
+  return (
+    <>
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/dashboard/components/RecentActivityAllProjects.jsx:45]: Don't need the empty wrapper

File: src/main/webapp/resources/js/pages/dashboard/components/RecentActivityList.jsx [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+ */
+export function RecentActivityList({
+  activities,
+  total,
+  page,
+  setPage,
+  loading,
+}) {
+  return (
+    <Space direction={"vertical"} style={{ width: `100%` }}>
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/dashboard/components/RecentActivityList.jsx:24]: Same comment here about the use of `Row`.  Prevents needing to add the `style={{width: "100%"}}` which I always thought was strange to add.

File: src/main/webapp/resources/js/pages/dashboard/components/RecentActivityUserProjects.jsx [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        const list = addKeysToList(data.content, "activity", "date");
+        setActivities([...activities, ...list]);
+        setTotal(data.total);
+        setLoading(false);
+      })
+      .catch((error) => notification.error({ message: error }));
+  }, [page]);
+
+  return (
+    <>
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/dashboard/components/RecentActivityUserProjects.jsx:45]: Don't need empty wrapper

**Review by joshsadam — CHANGES_REQUESTED** [review]
Overall, I love how this is going, just a few small changes.

File: src/main/webapp/resources/js/apis/activities/activities.js (hunk 5 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/apis/activities/activities.js:50]: The server returns a UserStatisticsResponse dto and not just a message. In this case there is no specific error message that would be returned. So I just added it here like we have for the project activity stuff

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIUserStatisticsService.java [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+@Component
+public class UIUserStatisticsService {
+
+	private UserService userService;
+	private ProjectService projectService;
+	private SampleService sampleService;
+	private AnalysisSubmissionService analysisSubmissionService;
+
+	public UIUserStatisticsService(UserService userService, ProjectService projectService, SampleService sampleService, AnalysisSubmissionService analysisSubmissionService) {
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIUserStatisticsService.java]: Fixed in 0042a0ad2e7a59b91ddeb0ff9efe5adaa46c6c25

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/users/dto/UserStatisticsResponse.java [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+		this.numberOfProjects = numberOfProjects;
+		this.numberOfSamples = numberOfSamples;
+		this.numberOfAnalyses = numberOfAnalyses;
+	}
+
+	public int getNumberOfProjects() {
+		return numberOfProjects;
+	}
+
+	public void setNumberOfProjects(int numberOfProjects) {
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/users/dto/UserStatisticsResponse.java]: Yup that's correct. Removed in 0042a0ad2e7a59b91ddeb0ff9efe5adaa46c6c25

File: src/main/java/ca/corefacility/bioinformatics/irida/service/impl/analysis/submission/AnalysisSubmissionServiceImpl.java (hunk 3 above)
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/service/impl/analysis/submission/AnalysisSubmissionServiceImpl.java]: Good catch. Updated in 0042a0ad2e7a59b91ddeb0ff9efe5adaa46c6c25

File: src/main/java/ca/corefacility/bioinformatics/irida/service/impl/ProjectServiceImpl.java (hunk 4 above)
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/service/impl/ProjectServiceImpl.java:511]: Will do!

File: src/main/webapp/resources/js/pages/dashboard/components/RecentActivity.jsx (hunk 6 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/dashboard/components/RecentActivity.jsx]: Yeah we need to figure it out and change it! Updated in 0042a0ad2e7a59b91ddeb0ff9efe5adaa46c6c25.

File: src/main/webapp/resources/js/pages/dashboard/components/RecentActivity.jsx (hunk 7 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/dashboard/components/RecentActivity.jsx]: Thanks! Yeah it works great. Updated in 0042a0ad2e7a59b91ddeb0ff9efe5adaa46c6c25

File: src/main/webapp/resources/js/pages/dashboard/components/RecentActivityAllProjects.jsx (hunk 8 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/dashboard/components/RecentActivityAllProjects.jsx]: Removed in 0042a0ad2e7a59b91ddeb0ff9efe5adaa46c6c25

File: src/main/webapp/resources/js/pages/dashboard/components/RecentActivityList.jsx (hunk 9 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/dashboard/components/RecentActivityList.jsx]: Updated in 0042a0ad2e7a59b91ddeb0ff9efe5adaa46c6c25. Yeah I found that to be strange as well

File: src/main/webapp/resources/js/pages/dashboard/components/RecentActivityUserProjects.jsx (hunk 10 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/dashboard/components/RecentActivityUserProjects.jsx]: Removed in 0042a0ad2e7a59b91ddeb0ff9efe5adaa46c6c25

**Review by joshsadam — APPROVED** [review]
This is such a massive improvement.  Thanks for tackling it :)

**Pull request merged**